flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/job/SubtaskAllExecutionAttemptsDetailsInfo.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.rest.messages.job;
+
+import org.apache.flink.runtime.rest.messages.ResponseBody;
+import org.apache.flink.util.Preconditions;
+
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonCreator;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonIgnore;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Collection;
+import java.util.Objects;
+
+/**
+ * The subtask all execution attempts response.
+ */
+public class SubtaskAllExecutionAttemptsDetailsInfo implements ResponseBody {
+	public static final String FIELD_NAME_ATTEMTPS = "attempts";
+
+	@JsonProperty(FIELD_NAME_ATTEMTPS)
+	private final Collection<SubtaskExecutionAttemptDetailsInfo> attempts;
+
+	@JsonCreator
+	public SubtaskAllExecutionAttemptsDetailsInfo(
+			@JsonProperty(FIELD_NAME_ATTEMTPS) Collection<SubtaskExecutionAttemptDetailsInfo> attempts) {
+		this.attempts = Preconditions.checkNotNull(attempts);
+	}
+
+	@JsonIgnore
+	public Collection<SubtaskExecutionAttemptDetailsInfo> getAttempts() {
+		return attempts;
+	}
+
+	@Override
+	public boolean equals(Object o) {
```

Comment:
Are we adding `equals(Object)` to facilitate testing? This is generally not considered a good practice:
* https://softwareengineering.stackexchange.com/questions/337533/is-it-considered-a-bad-practice-to-create-the-object-equals-method-for-testing?noredirect=1&lq=1
* https://stackoverflow.com/questions/1180044/should-one-override-equals-method-for-asserting-the-object-equality-in-a-unit-te

Reply:
I added `equals(Object)` as all object for rest API need to override `equals` and `hashCode`